transport-native-epoll/src/main/c/netty_epoll_native.c

```diff
@@ -315,7 +315,7 @@ static jstring netty_epoll_native_kernelVersion(JNIEnv* env, jclass clazz) {
 }
 
 static jboolean netty_epoll_native_isSupportingSendmmsg(JNIEnv* env, jclass clazz) {
-    if (sendmmsg) {
+    if (&sendmmsg) {
```

Comment:
> But this solution doesn't work. The symbol is not weakly defined. If IO_NETTY_SENDMMSG_NOT_FOUND is set, it's a compile time error for redefining it:

I'm not suggesting redefining anything. The purpose of `IO_NETTY_SENDMMSG_NOT_FOUND` is to avoid this and should remain.

> The real issue is that this file is selectively defining the method as weak, which is the true problem

This is the approach taken to supporting compiling on pre 3.0 kernels, post 3.0 kernels, and linking artifacts built on pre 3.0 kernels on post 3.0 kernels (and visa versa). Are you aware of an alternative approach?

> Did you try compiling with -Wpointer-bool-conversion ?

I think the warning in question is `-Wtautological-pointer-compare`? We should be explicitly checking against `NULL` instead of the implicit boolean check either way. clang actually gives the below recommendation ... this warning seems to be a bit overzealous in this case do to the linking nature of the problem but I guess it shouldn't hurt (assuming the address of operator is always safe the function isn't defined ... as I said seems to be this way in practice but I would be interested in a link which explains in more detail)

```
scott@scott-lubuntu:~$ clang -Werror test.c -o test
test.c:14:10: error: comparison of function 'sendmmsg' not equal to a null pointer is always true [-Werror,-Wtautological-pointer-compare]
     if (sendmmsg != NULL) {
         ^~~~~~~~    ~~~~
test.c:14:10: note: prefix with the address-of operator to silence this warning
     if (sendmmsg != NULL) {
         ^
         &
```